tls/s2n_protocol_preferences.c

```diff
@@ -108,7 +108,10 @@ S2N_RESULT s2n_protocol_preferences_set(struct s2n_blob *application_protocols,
      */
     RESULT_GUARD_POSIX(s2n_realloc(&new_protocols, 0));
 
-    for (size_t i = 0; i < protocol_count; i++) {
+    /* Check to ensure protocol_count is non-negative */
+    RESULT_ENSURE_GTE(protocol_count, 0);
+
```

Comment:
nit: Same comment on the comment. If the check comes right before where you cast it, it might be clear enough without the comment.
```suggestion
    RESULT_ENSURE_GTE(protocol_count, 0);
```